libmusig2.h

```diff
@@ -1,69 +1,196 @@
-                            const unsigned char *partial_sigs, const unsigned char *announcement,
-                            const unsigned char *m, unsigned long long mlen, unsigned long nr_signers);
-
-#endif //MUSIG2_LIBRARY_H
+#include <stdlib.h>
+#include <stdio.h>
+#include <assert.h>
+#include <secp256k1.h>
+#include "random.h"
+
+/*************************************************************************/
+/****************************** DEFINITIONS ******************************/
+/*************************************************************************/
+#define N               7 /* Number of signers */
+
+/************ DO NOT CHANGE ************/
+#define V               2 /* Number of nonce values. Note that V>2 is not working yet. */
+
+#define SK_BYTES        32
+#define SK_SIZE	        SK_BYTES*sizeof(unsigned char)
+
+#define PK_BYTES        33
+#define PK_SIZE         PK_BYTES*sizeof(unsigned char)
+/***************************************/
+/*************************************************************************/
+
+
+/*************************************************************************/
+/************************* DATA STRUCTURES *******************************/
+/*************************************************************************/
+/* Pointer:     SIGNER
+ * Purpose:     Pointer to a signer object holding signer keys and nonce values.
+ * Parameters:  sk: Signer's secret key, 32 Bytes.
+ *              r:  Char array that holds V random values of size SK_SIZE.
+ *              PK: Signer's compressed public key of size 33 Bytes.
+ */
+typedef struct{
+    unsigned char* sk;
+    unsigned char** r ;
+    unsigned char* PK;
+    unsigned char** R;
+}SIGNER_t[0], *SIGNER;
+
+/* Type:        NONCE
+ * Purpose:     A type storing the nonce values of signers.
+ * Parameters:  R : Char array holding N nonce values of size PK_SIZE.
+ */
+struct NONCE{
+    unsigned char** R;
+};
+
+/* Pointer:     AGG_ROUND1
+ * Purpose:     Pointer to an object holding the parameters and output of first round signature.
+ * Parameters:  PK_LIST:    The list of public keys.
+ *              R_LIST:     An array type of NONCE that holds V NONCE.
+ *              OUT_R_LIST: The aggregated output of first round.
+ *              OUT:        The concatenation of the content of OUT_R_LIST.
+ */
+typedef struct{
+    unsigned char** PK_LIST;
+    struct NONCE** R_LIST;
+    unsigned char** OUT_R_LIST;
+    unsigned char* OUT;
+}AGG_ROUND1_t[0], *AGG_ROUND1;
+
+/* Pointer:     PARAM_ROUND2
+ * Purpose:     Pointer to an object holding the parameters of the second round signature.
+ * Parameters:  X_:         The aggregated public key.
+ *              L:          Concatenation of public keys.
+ *              b:          Stores H(X_, OUT, msg).
+ *              R_state:    Stores the state of second round R = SUM(Rj*b^j).
+ */
+typedef struct{
+    unsigned char* X_;
+    unsigned char* L;
```

Comment:
L shouldn't be in round2 parameters, but instead in a global parameter structure.